From patchwork Fri Jan 25 07:32:46 2019
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Masahiro Yamada <yamada.masahiro@socionext.com>
X-Patchwork-Id: 10780731
Return-Path: <kvm-owner@kernel.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 946AE13B5
	for <patchwork-kvm@patchwork.kernel.org>;
 Fri, 25 Jan 2019 07:34:12 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 767972C148
	for <patchwork-kvm@patchwork.kernel.org>;
 Fri, 25 Jan 2019 07:34:12 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id 69E102C20F; Fri, 25 Jan 2019 07:34:12 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-7.9 required=2.0 tests=BAYES_00,DKIM_SIGNED,
	DKIM_VALID,MAILING_LIST_MULTI,RCVD_IN_DNSWL_HI autolearn=ham version=3.3.1
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 56D6A2C148
	for <patchwork-kvm@patchwork.kernel.org>;
 Fri, 25 Jan 2019 07:34:11 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1727264AbfAYHeF (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Fri, 25 Jan 2019 02:34:05 -0500
Received: from conuserg-10.nifty.com ([210.131.2.77]:33570 "EHLO
        conuserg-10.nifty.com" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S1726271AbfAYHeF (ORCPT <rfc822;kvm@vger.kernel.org>);
        Fri, 25 Jan 2019 02:34:05 -0500
Received: from pug.e01.socionext.com (p14092-ipngnfx01kyoto.kyoto.ocn.ne.jp
 [153.142.97.92]) (authenticated)
        by conuserg-10.nifty.com with ESMTP id x0P7WwQf018950;
        Fri, 25 Jan 2019 16:32:58 +0900
DKIM-Filter: OpenDKIM Filter v2.10.3 conuserg-10.nifty.com x0P7WwQf018950
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=nifty.com;
        s=dec2015msa; t=1548401579;
        bh=tbbuiCD7IbfQSxtcRUBQrQMr2BXXvMyScGRF6PcyoDI=;
        h=From:To:Cc:Subject:Date:From;
        b=Fs7U81ZlT9/ijMJkA921Qgf2+hkeTHQPFvNKWazjFiV2i64ReRYAgaY1URp4xx3+U
         oRXca2uYQbWPpnKUMdsy2Huck4jER7fP0BGMUG6amKoM3Y4Kk3ueOSi+Bs769SO07i
         IG1jsTx4+yZEKRnInnT0U6S9cxmShjGpkHHDYxsAwQ0vJTk2pV0cLxDp8FAlN/3TDd
         ycbz9ckYkq1+VvivmVM6coI+gs8f+RA4ir1Kf7XB/Y0Ibtn7hZq9x8Bc5o5uRr0jpO
         ewfK8yc96SpwcS4B9mYXmOch1kBsJ1TUEqpahZYUfVxFC1w/6tAJXTXdKBJaG6U8UT
         yFlEJp40zbxjA==
X-Nifty-SrcIP: [153.142.97.92]
From: Masahiro Yamada <yamada.masahiro@socionext.com>
To: =?unknown-8bit?b?UGFvbG8gQm9uemluaSA8cGJvbnppbmlAcmVkaGF0LmNvbT4sIFJh?=
	=?unknown-8bit?b?ZGltIEtyj6utbY+roY+r2iAgIDxya3JjbWFyQHJlZGhhdC5jb20+LCBr?=
	=?unknown-8bit?b?dm1Admdlci5rZXJuZWwub3Jn?=
Cc: Masahiro Yamada <yamada.masahiro@socionext.com>,
        "H. Peter Anvin" <hpa@zytor.com>, Borislav Petkov <bp@alien8.de>,
        x86@kernel.org, linux-kernel@vger.kernel.org,
        Thomas Gleixner <tglx@linutronix.de>,
        Ingo Molnar <mingo@redhat.com>
Subject: [PATCH] KVM: x86: fix TRACE_INCLUDE_PATH and remove -I. header search
 paths
Date: Fri, 25 Jan 2019 16:32:46 +0900
Message-Id: <1548401566-29835-1-git-send-email-yamada.masahiro@socionext.com>
X-Mailer: git-send-email 2.7.4
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org
X-Virus-Scanned: ClamAV using ClamSMTP

The header search path -I. in kernel Makefiles is very suspicious;
it allows the compiler to search for headers in the top of $(srctree),
where obviously no header file exists.

The reason of having -I. here is to make the incorrectly set
TRACE_INCLUDE_PATH working.

As the comment block in include/trace/define_trace.h says,
TRACE_INCLUDE_PATH should be a relative path to the define_trace.h

Fix the TRACE_INCLUDE_PATH, and remove the iffy include paths.

Signed-off-by: Masahiro Yamada <yamada.masahiro@socionext.com>
---

 arch/x86/kvm/Makefile | 4 ----
 arch/x86/kvm/trace.h  | 2 +-
 2 files changed, 1 insertion(+), 5 deletions(-)

diff --git a/arch/x86/kvm/Makefile b/arch/x86/kvm/Makefile
index 69b3a7c..31ecf7a 100644
--- a/arch/x86/kvm/Makefile
+++ b/arch/x86/kvm/Makefile
@@ -2,10 +2,6 @@
 
 ccflags-y += -Iarch/x86/kvm
 
-CFLAGS_x86.o := -I.
-CFLAGS_svm.o := -I.
-CFLAGS_vmx.o := -I.
-
 KVM := ../../../virt/kvm
 
 kvm-y			+= $(KVM)/kvm_main.o $(KVM)/coalesced_mmio.o \
diff --git a/arch/x86/kvm/trace.h b/arch/x86/kvm/trace.h
index 705f40a..6432d08 100644
--- a/arch/x86/kvm/trace.h
+++ b/arch/x86/kvm/trace.h
@@ -1465,7 +1465,7 @@ TRACE_EVENT(kvm_hv_send_ipi_ex,
 #endif /* _TRACE_KVM_H */
 
 #undef TRACE_INCLUDE_PATH
-#define TRACE_INCLUDE_PATH arch/x86/kvm
+#define TRACE_INCLUDE_PATH ../../arch/x86/kvm
 #undef TRACE_INCLUDE_FILE
 #define TRACE_INCLUDE_FILE trace
 
